src/utils/chat-utils.ts

```diff
@@ -150,23 +155,22 @@ export async function getResponse(
  * Autocompletes a JSON string with default values for the keys specified in defaultValues.
  * @param incompleteJSON The possibly incomplete JSON string to be autocompleted streamed from the server.
  * @param defaultValues A set of keys to be written into the JSON string if they are not already present.
- * @returns a completed JSON object, specifically in the format of a ChatResponseStream object.
+ * @returns a completed JSON object.
  */
-function completeJSON(incompleteJSON: string, defaultValues: string[]): ChatResponseStream {
+function completeJSON(incompleteJSON: string, defaultValues: string[]): {[k: string]: string} {
     const jsonStr = untruncateJson(incompleteJSON);
-    let res = undefined;
+    let res = {};
     try {
         res = JSON.parse(jsonStr);
     } catch (e) {
         console.warn('Error parsing JSON', e);
```

Comment:
Am I wrong in guessing that this generates a lot of spam to the console?

Reply:
Were you referring to this line 
```js
console.warn('Error parsing JSON', e);
```
?
I am open to removing it.